### PR TITLE
Add "user sign up" endpoint

### PR DESCRIPTION
**Pull request by jortizsao**
* Adds an endpoint that given a user draft, returns the new customer and a valid JWT token.
* Adds fastest-validator library for validating the input params

### CODE REVIEW

File: server/src/validator/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        message: `${previous.message}${previous.message ? ', ' : ''}${vr.message}`,
+        errors: [...previous.errors, { type, field, message }],
+      };
+    },
+    { message: '', errors: [] },
+  );
+
+export default schema => (req, res, next) => {
+  const validator = new Validator();
+  const check = validator.compile(schema);
```
Comment (lojzatran) [inline server/src/validator/index.js]: Should you first compile schema and then just use `check` function? Here you compile every time this function is called?

File: server/src/api/auth/specs/auth.controller.spec.js [hunk 2]
```diff
@@ -80,4 +84,55 @@ describe('Auth controller', () => {
...
+        req.body = {
+          firstName: 'dummy-name',
+          lastName: 'dummy-last-name',
+          email: 'dummy-email@commercetools.de',
+          password: '12345',
+        };
+
+        usersService.create = jest.fn().mockReturnValue(Promise.resolve(newUser));
+
+        authController.signUp(req, res);
```
Comment (lojzatran) [inline server/src/api/auth/specs/auth.controller.spec.js]: await?

File: server/src/validator/index.js (hunk 1 above)
Comment (jortizsao) [reply server/src/validator/index.js]: it's a high order function. it's just compiled once (when the validator instance is created) and returns a middleware (another function) that Express will use on each request that actually validates the params using the previous schema thanks to the closure

File: server/src/api/auth/auth.service.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ export default ({ commercetools }) => {
           throw new NotAuthenticatedError();
         }
 
-        throw err;
+        throw new CommercetoolsError(err);
```
[the remark this answers is not in the document]
Comment (jortizsao) [reply server/src/api/auth/auth.service.js]: the error comes from the commercetools-sdk, that's why I use a CommercetoolsError.

what other error type should I use? how do I differentiate the errors that come from the commercetools sdk?

File: server/src/validator/index.js (hunk 1 above)
Comment (jortizsao) [reply server/src/validator/index.js]: Sorry, @lojzatran I notice that you are right! :) the high order function is not correctly set. the compilation of the schema should be out of the second function. I change that! Thanks!

File: server/src/api/auth/auth.service.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lojzatran) [reply server/src/api/auth/auth.service.js]: Maybe:
if (err.statusCode)
  // ctp error
  throw new CommercetoolsError(err)
else
  // different error
  throw err

[the remark this answers is not in the document]
Comment (jortizsao) [reply server/src/api/auth/auth.service.js]: https://github.com/commercetools/kubernetes-custom-ui/pull/21/commits/5d047ecf6c0d982fb777e9793e9fb171df9d126a

File: server/src/validator/index.js (hunk 1 above)
Comment (jortizsao) [reply server/src/validator/index.js]: https://github.com/commercetools/kubernetes-custom-ui/pull/21/commits/5d047ecf6c0d982fb777e9793e9fb171df9d126a

File: server/src/api/auth/specs/auth.controller.spec.js (hunk 2 above)
Comment (jortizsao) [reply server/src/api/auth/specs/auth.controller.spec.js]: https://github.com/commercetools/kubernetes-custom-ui/pull/21/commits/5d047ecf6c0d982fb777e9793e9fb171df9d126a

**Pull request merged**